website/blog/modules/ROOT/pages/15-android-build-flow.adoc

```diff
@@ -0,0 +1,223 @@
+= How does Mill build Android apps?
+
+// tag::header[]
+:author: Vasilis Nicolaou
+:revdate: 29 August 2025
+
+_{author}, {revdate}_
+
+
+Until recently, Gradle was the only realistic option for Android builds. Today, Mill - a fast, predictable build tool for Java, Scala, and Kotlin - can build Android applications end-to-end.
+
+In less than a year, Mill went from minimal Android support to producing installable APKs for projects as complex as xref:mill::android/hilt-sample.adoc[the Android Architecture Samples] and https://github.com/vaslabs/Pokedex_Compose_Multi_Module/tree/testing-mill[multi-module apps] with xref:mill::android/java.adoc#_using_third_party_native_libraries[native code] and xref:mill::android/compose-samples.adoc[Jetpack Compose].
+// end::header[]
+Multi-module Pokedex sample app (https://github.com/NicosNicolaou16/Pokedex_Compose_Multi_Module[original], https://github.com/vaslabs/Pokedex_Compose_Multi_Module/tree/testing-mill[with mill])
+image:AndroidPokedexMultimoduleExample.png[A multi-module Android app built with Mill, showing a list of Pokémon and details for each pokemon.]
+
+Unlike Gradle, where build steps are buried inside plugins and DSL magic, Mill exposes each Android step as a task you can run, override, or extend.
```

Comment:
Is there any angle we can discuss of Mill being easier to setup v.s. Gradle in terms of the number of manual steps? Or more performant, in terms of time-to-update-the-app? We should try to discuss some user-facing benefits in a way people can see for themselves why it's cool, rather than just "Mill is fast and predictable because we say so", so they are more engaged for the internal technical discussion below